BootloaderCorePkg/Tools/PrepareBuildComponentBin.py

```diff
@@ -212,122 +225,32 @@ def BuildFspBins (fsp_dir, sbl_dir, platform, flag):
-    if not os.path.exists(ucode_dir + '/.git'):
-        print ('Cloning Intel Ucode repo ...')
-        cmd = 'git clone https://github.com/intel/Intel-Linux-Processor-Microcode-Data-Files.git %s' % ucode_dir
-        ret = subprocess.call(cmd.split(' '))
-        if ret:
-            Fatal ('Failed to clone Ucode repo to directory %s !' % ucode_dir)
-        print ('Done\n')
-    else:
-        print ('Update Intel Ucode repo ...')
-        cmd = 'git pull origin master'
-        ret = subprocess.call(cmd.split(' '), cwd=ucode_dir)
-        if ret:
-            Fatal ('Failed to update Ucode repo to directory %s !' % ucode_dir)
-        print ('Done\n')
-
-    print ('Checking out Intel Ucode commit ...')
-    cmd = 'git checkout %s' % tag
-    ret = subprocess.call(cmd.split(' '), cwd=ucode_dir)
-    if ret:
-        Fatal ('Failed to check out tag %s !' % tag)
-
-def GetUcodeList (platform):
-    plat_pkg  = GetPlatformPkgName (platform)
-    ucode_inf = os.path.join(os.environ['PLT_SOURCE'], 'Silicon/%s/Microcode/Microcode.inf' % plat_pkg)
-
-    fd = open (ucode_inf, 'r')
-    lines = fd.readlines()
-    fd.close ()
-
-    ucode_tag      = ''
-    is_def_section = False
-    is_src_section = False
-    ucode_names    = []
-    for line in lines:
-        line = line.strip ()
-        if line.startswith('['):
-            if line.upper().startswith('[DEFINES]'):
-                is_def_section = True
-            else:
-                is_def_section = False
-            if line.upper().startswith('[SOURCES]'):
-                is_src_section = True
-            else:
-                is_src_section = False
-
-        if is_def_section:
-            match = re.match("^MICROCODE_TAG\s*=\s*([_\-\w]+)", line)
-            if match:
-                ucode_tag = match.group(1)
-
-        if is_src_section:
-            match = re.match("^([_\-\w]+)\.mcb", line)
-            if match:
-                ucode_names.append(match.group(1))
-
-    if ucode_tag == '':
-        ucode_tag = 'microcode-20190514a'
-
-    if len(ucode_names) == 0:
-        Fatal ('Failed to find microcode files in file - %s' % ucode_inf)
-
-    ucode_list = []
-    for ucode in ucode_names:
-        ucode_list.append (('intel-ucode/%s' % ucode, 'Silicon/%s/Microcode/%s.mcb' % (plat_pkg, ucode)))
-
-    return ucode_list, ucode_tag
-
-def CopyUcodeBins (ucode_dir, sbl_dir, platform):
-    if platform not in ['cfl']:
-        return
-
-    copy_list, ucode_tag = GetUcodeList (platform)
-    if CheckFileListExist(copy_list, sbl_dir):
-        return
-
-    CloneUcodeRepo (ucode_dir, ucode_tag)
-
-    CopyFileList (copy_list, ucode_dir, sbl_dir)
-
-    print ('Done\n')
+    CopyFileList (copy_list, fsp_dir)
 
 def Main():
-    curr_dir       = os.path.dirname (os.path.realpath(__file__))
-    workspace_dir  = os.path.abspath (os.path.join(curr_dir, '../../..'))
+    sbl_dir        = os.environ['PLT_SOURCE']
+    workspace_dir  = os.path.join(sbl_dir, 'download')
```

Comment:
let us not change the behavior here.  All downloaded repo should be located outside of current SBL tree.

Reply:
Will move to upper level for SblOpen. the idea is to have single download folder to save different downloaded repos.

Reply:
BTW, can we use "Download" to be consistent with the naming convention ?

Reply:
Done